cassiopeia/solver/DistanceSolver.py

```diff
@@ -7,14 +7,14 @@
 """
 import abc
 import cassiopeia
+from cassiopeia.data import CassiopeiaTree
+from cassiopeia.mixins import DistanceSolverError
+from cassiopeia.solver import CassiopeiaSolver, solver_utilities
```

Comment:
Sorry this isn't necessarily right. The import structure should separate the third party libraries (e.g., networkx) from our Cassipoeia libraries.

It should be: 

```
from typing import Callable, Dict, List, Optional, Tuple

import itertools
import networkx as nx
import numpy as np
import pandas as pd
import scipy

from cassiopeia.data import CassiopeiaTree
from cassiopeia.solver import (
    dissimilarity_functions,
    DistanceSolver,
    DistanceSolver,
    solver_utilities,
)

```

`typing` goes at the top because it's actually part of the base python distribution.